artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ScheduledDeliveryHandlerImpl.java

```diff
@@ -135,6 +134,10 @@ public List<MessageReference> cancel(final Filter filter) throws ActiveMQExcepti
       return refs;
    }
 
+   public interface CancelComparator {
+      boolean compare(MessageReference ref) throws ActiveMQException;
```

Comment:
The ActiveMQException isn't required if the the filter is created before calling the comparator and the CancelComparator could be replaced by java.util.function.Predicate<MessageReference>

Reply:
This would be a good improvement. Thanks!